frontend/src/app/components/search.tsx

```diff
@@ -1,13 +1,59 @@
+"use client"
+
 import { Button } from "@/components/ui/button"
 import { Input } from "@/components/ui/input"
- 
+import { Autocomplete, useJsApiLoader } from "@react-google-maps/api";
+import { useState } from "react";
+import { useGoogleMapsLoader } from "./useGoogleMapsLoader";
+
 function SearchBar() {
+  const [searchResult, setSearchResult] = useState<google.maps.places.Autocomplete>()
+  const center = { lat: 50.064192, lng: -130.605469 };
+  const defaultBounds = {
+    north: center.lat + 0.1,
+    south: center.lat - 0.1,
+    east: center.lng + 0.1,
+    west: center.lng - 0.1,
+  };
+  const options = {
+    bounds: defaultBounds,
+    componentRestrictions: { country: "us" },
+    fields: ["address_components", "geometry", "icon", "name", "place_id,  adr_address"],
+    strictBounds: false,
+  };
+
+  const { googleMaps, isLoaded, loadError } = useGoogleMapsLoader();
+
+  function onLoad(autocomplete:google.maps.places.Autocomplete ) {
+    setSearchResult(autocomplete);
+  }
+
+  function onPlaceChanged() {
+    if (searchResult != null) {
+      const place = searchResult.getPlace();
+      const name = place.name;
+      const status = place.business_status;
+      const formattedAddress = place.formatted_address;
+      console.log(`Name: ${name}`);
```

Comment:
Do we want to keep these `console.log` statements here?

Reply:
For now probably, its helpful to understand the calls to the api.